addons/mrp/models/mrp_production.py

```diff
@@ -1418,27 +1418,31 @@ def _cal_price(self, consumed_moves):
         return True
 
     def _post_inventory(self, cancel_backorder=False):
+        moves_not_to_do = self.move_raw_ids.filtered(lambda x: x.state == 'done')
+        moves_to_do = self.move_raw_ids.filtered(lambda x: x.state not in ('done', 'cancel'))
+        for move in moves_to_do.filtered(lambda m: m.product_qty == 0.0 and m.quantity_done > 0):
+            move.product_uom_qty = move.quantity_done
```

Comment:
don't know if this is really significant but we could replace those 3 filtered's and the loop by only one loop on self.move_raw_ids

Reply:
@Whenrow Yeah good idea. From my tests in local it seems to speedup the 658 mo case by around 10-15s.